### PR TITLE
Implement MS16007: Add support for PNG snapshots

### PR DESCRIPTION
**Pull request by zfox23**
Implements [MS16007](https://highfidelity.fogbugz.com/f/cases/16007/) for @samcake and anyone else who wants to take lossless Snapshots.

You can now save snapshots in the `png` format by specifying a `.png` filename in the `filename` argument to your `Window.takeSnapshot()`, `Window.takeSecondaryCameraSnapshot()`, or `Window.takeSecondaryCamera360Snapshot()` calls.

---

Hoo boy, looks like Clang had some fun with this file while I was editing it. My only changes are on lines 53, and 363-378.

### CODE REVIEW

**Comment by zfox23** [comment]
### Test Plan
1. Connect to a domain. Set up your Snapshot directory. Open the Scripting Console (`CTRL+ALT+J`). Type the following and press enter: `Window.takeSnapshot(true, false, 0.0, "test.png")`
    - Verify that you now have a **PNG Snapshot** called `test.png` in your Snapshot directory.
2. Type the following in the Scripting Console and press enter: `Window.takeSnapshot(true, false, 0.0, "test.jpeg")`
    - Verify that you now have a **JPG Snapshot** called `test.jpeg` in your Snapshot directory.
3. Type the following in the Scripting Console and press enter: `Window.takeSnapshot(true, false, 0.0, "test.jpg")`
    - Verify that you now have a **JPG Snapshot** called `test.jpg` in your Snapshot directory.
4. Type the following in the Scripting Console and press enter: `Window.takeSnapshot(true, false, 0.0, "test.gif")`
    - Verify that you now have a **JPG Snapshot** called `test.gif.jpg` in your Snapshot directory.
5. Open the Spectator Camera app (get it from the Marketplace if you don't have it). Turn on the Spectator Camera. Type the following in the Scripting Console and press enter: `Window.takeSecondaryCameraSnapshot(true, "test2.png")`
    - Verify that you now have a **PNG Snapshot** called `test2.png` in your Snapshot directory.
6. Type the following in the Scripting Console and press enter: `Window.takeSecondaryCamera360Snapshot(JSON.stringify("{x: 0, y: 0, z: 0}"), false, true, "test3.png")`
    - Verify that you now have a **360 PNG Snapshot** called `test3.png` in your Snapshot directory.

File: interface/src/ui/Snapshot.cpp [hunk 1]
```diff
@@ -350,12 +360,22 @@ QFile* Snapshot::savedFileForSnapshot(QImage & shot, bool isTemporary, const QSt
 
     QDateTime now = QDateTime::currentDateTime();
 
-    // If user has requested specific filename then use it, else create the filename
-	// 'jpg" is appended, as the image is saved in jpg format.  This is the case for all snapshots
-	//       (see definition of FILENAME_PATH_FORMAT)
+    // If user has supplied a specific filename for the snapshot:
```
Comment (samcake) [inline interface/src/ui/Snapshot.cpp]: could you copy that explanationover in the header for the function so it goes into the documentation

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10672/HighFidelity-Beta-PR13405-dbf3ade.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10675/HighFidelity-Beta-PR13405-e83b767.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10675/HighFidelity-Beta-PR13405-e83b767.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10675/HighFidelity-Beta-Interface-PR13405-e83b767.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10675/HighFidelity-Beta-PR13405-e83b767.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10675/HighFidelity-Beta-Interface-PR13405-e83b767.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
PNG and JPG files are created successfully.

**Pull request merged**